ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -551,6 +803,7 @@ public class Manager: NSObject, HTTPDownloadDelegate, UIGestureRecognizerDelegat
       downloadQueue!.addRequest(request)
     }
     downloadQueue!.run()
+    self.downloadLexicalModelsForLanguageIfExists(languageID: languageID)
```

Comment:
I think you should add the URL to the lexical model to the `downloadQueue` before it gets run in `downloadQueue!.run()`. This way, it will download all the keyboard JS files, fonts, and lexical model files for you.

Probably `downloadQueue!.addRequest(request)` where `request` is updated for lexical model info.

Reply:
psuedocode
```
if (associatedLexicalModel) {
   request = HTTPDownloadRequest(url_to_lexical_model, userInfo)
   request.destinationFile = Storage....
   request.tag = i+1 (increment from keyboard and/or fonts)
   downloadQueue!.addRequest(request)
}
```

Reply:
That would work for a simple file, but I need to download the list  of  models,  then pick  one (for  now the  first),  then download and process it.
I've finally done it.

Reply:
You don't download the list of models.
From the lexical models API query, you get a list of models returned. From that list, you can determine the first model's `packageFilename`. This URL gets added to the downloadQueue.

Reply:
It doesn't look like we're relying on the download queue to process the API query, so it does seem that what Darcy's suggesting is possible.  (Refer to [line 32 of APILexicalModelRepository.swift](https://github.com/keymanapp/keyman/pull/1707/files?file-filters%5B%5D=.md&file-filters%5B%5D=.plist&file-filters%5B%5D=.resolved&file-filters%5B%5D=.swift&file-filters%5B%5D=No+extension#diff-fde50d0047ffef23214ca374e58ed607R25)).  I'm not sure how much value it adds at the moment to require a refactor of this section, though, so long as it's eventually downloaded through the queue.